### PR TITLE
[merged] sysroot: Drop an unnecessary fsync

### PR DESCRIPTION
**Pull request by cgwalters**
While looking at a slow update issue (which I'm guessing is
unpredictable I/O latency in an OpenStack instance), I noticed
in one of the traces we were inside a fsync here.

Dropping the fsync here is just another of a long series of unwinding
them - we `syncfs()` the sysroot fd and `/boot` and we have a big
`sync()` anyways.

### CODE REVIEW

**Comment by cgwalters** [comment]
Rebased :surfer:

File: src/libostree/ostree-sysroot-deploy.c [hunk 1]
```diff
@@ -1827,9 +1827,10 @@ _ostree_sysroot_write_deployments_internal (OstreeSysroot     *self,
     {
       int new_bootversion = self->bootversion ? 0 : 1;
```
Comment (jlebon) [inline src/libostree/ostree-sysroot-deploy.c]: Don't mind me, just testing the new review feature!

File: src/libotutil/ot-gio-utils.c [hunk 2]
```diff
@@ -315,64 +315,28 @@ ot_gfile_ensure_unlinked (GFile         *path,
 }
 
 /**
- * ot_util_ensure_directory_and_fsync:
+ * ot_util_fsync_directory:
```
Comment (jlebon) [inline src/libotutil/ot-gio-utils.c]: Why not just remove it? Are we planning to make use of it in the future?

Comment (cgwalters) [inline src/libotutil/ot-gio-utils.c]: Yep, it was a bad merge.  Fixed.

**Comment by jlebon** [comment]
I guess review feedback doesn't show up as comments in the API. 
@rh-atomic-bot r+ ab068df

**Pull request closed without merging**